### PR TITLE
Fix: Deterministic order when computing fingerprint for jinja macro packages

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

**Review by crericha — APPROVED** [review]

File: sqlmesh/core/snapshot/definition.py [hunk 1]
```diff
@@ -942,7 +942,7 @@ def _model_data_hash(model: Model) -> str:
             data.append(macro_name)
             data.append(macro.definition)
 
-        for package in model.jinja_macros.packages.values():
+        for _, package in sorted(model.jinja_macros.packages.items(), key=lambda x: x[0]):
```
Comment (vchan) [inline sqlmesh/core/snapshot/definition.py:945]: Nit: The key argument isn't really needed here.

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/snapshot/definition.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/snapshot/definition.py:945]: what do you mean?

Comment (izeigerman) [reply sqlmesh/core/snapshot/definition.py:945]: we need it for sorting

Comment (vchan) [reply sqlmesh/core/snapshot/definition.py:945]: I don't think it's a big deal. I meant `sorted` will use the first element of x automatically for sorting.

Comment (tobymao) [reply sqlmesh/core/snapshot/definition.py:945]: big brain

```python
>>> sorted({"b": 1, "a": 2}.items())
[('a', 2), ('b', 1)]
```

**Pull request merged**